uint8 FULL=0
uint8 SLOW=1
uint8 STOP=2
uint8 UNKNOWN=3
uint8 current_state
